addon/services/intl.js

```diff
@@ -38,7 +40,7 @@ export default Service.extend(Evented, {
 
       if (!isArrayEqual(proposed, this._locale)) {
         this._locale = proposed;
-        this.trigger('localeChanged');
+        next(() => this.trigger('localeChanged'));
```

Comment:
why this change?

Reply:
The locale does not actually change until _after_ this setter has run through. So emitting this event _during_ the setter causes all (synchronous) listeners accessing the `locale` property to receive the old value.

The reason that it works in the `{{t}}` helper is that [`recompute`](https://github.com/emberjs/ember.js/blob/v3.5.0/packages/ember-glimmer/lib/helper.ts#L90-L119) marks the helper as dirty and it's only recomputed later in a different run loop queue.

https://github.com/ember-intl/ember-intl/blob/89009718cb36ebfce0b2783e9aff05702d9f6431/addon/helpers/-format-base.js#L22

I only noticed this "bug" now, because I used the event further down in the class.

Reply:
this should be an independent bugfix PR then, right? I guess it would also be good to have some sort of test for this so that we don't regress on this again.

Reply:
I'll pull it out. 👌